### DISCUSSION TITLE
Change settings.py?

### DISCUSSION THREAD

**Discussion by j4nd3r53n** [inventree/InvenTree#4464] [General]
I have installed bare metal InvenTree on Debian 11 and now need to edit the `settings.py` - it seems there are three such files:

```
root@vogon:~# find / /var -xdev -name settings.py
/InvenTree/InvenTree/common/settings.py
/InvenTree/InvenTree/part/settings.py
/InvenTree/InvenTree/InvenTree/settings.py
```

I think the one I need to change is probably `/InvenTree/InvenTree/InvenTree/settings.py`, but at the top it says:

```
In practice the settings in this file should not be adjusted,
instead settings can be configured in the config.yaml file
located in the top level project directory.
```

A search for `config.yaml` (or `config.yml`) hassn't found anything - is it ok to simply make changes in the `settings.py`? Or is there somewhere in the Web interface to do it?

What I want to do is add LDAP authentication (I know there are better alternatives and so on, but that isn't an option), following the instructions for `django-auth-ldap` as exemplified [here](https://django-auth-ldap.readthedocs.io/en/latest/example.html), which seem reasonably understandable.

**Comment by matmair** [comment]
First of all: I really do not recommend changing the settings. And I do not recommend LDAP - we will not support problems with it.

You can change the settings file you pointed out. Automated updates will not work. You probably will have to fix the diffs from time to time. We will probably have problems identifying the specific version you have on bug reports because the SHAs do not match. Please follow https://github.com/inventree/InvenTree/issues/3746 to see when you can switch to upstream InvenTree.

The `config.yaml` gets generated at the first startup of the server / worker.

**Reply by j4nd3r53n** [reply]
Thank you for replying! I understand that you won't touch LDAP even with a long barge pole - neither would I, but we don't always get the luxury of choosing.

My real question was about how best to change the settings; I have no experience with django, only Python. We have been using InvenTree for a month, so I assume `config.yaml` should have been generated already,  but I haven't found it. However, if it is generated, then presumably it isn't the source for `settings.py`?

**Reply by SchrodingersGat** [reply]
The path for `config.yaml` is determined here:

https://github.com/inventree/InvenTree/blob/2dfea9b825e3b742b48a84acd5d3688ba418f222/InvenTree/InvenTree/config.py#L78
